File: python3/power_of_two.py
# leetcode

class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n <= 0:
            return False 
        ones = 0
        
        binary = bin(n)[2:]
        
        for i in binary:
            if i == "1":
                ones += 1
        
        if ones == 1:
            return True
        return False
        

File: golang/power_of_two.go
package main

import "fmt"

// Space	: O(1)
// Time		: O(n)

func isPowerOfTwo(n int) bool {
	if n < 0 {
		return false
	}

	b := fmt.Sprintf("%b", n)
	ones := 0

	for i := 0; i < len(b); i++ {
		if b[i] == 49 {
			ones++
		}
	}

	return ones == 1
}
